tokio/src/time/instant.rs

```diff
@@ -0,0 +1,187 @@
+#![allow(clippy::trivially_copy_pass_by_ref)]
+
+use std::fmt;
+use std::ops;
+use std::time::Duration;
+
+/// A measurement of the system clock, useful for talking to
+/// external entities like the file system or other processes.
```

Comment:
Some questions:

- why do we have our own `Instant` now? is there a reason we can't use `std::time::Instant`?
- can this doc comment explain why this type exists and how it differs from `std::time::Instant`?
- if this adds new methods, is there a rationale for a new type vs. an extension trait?

Reply:
https://github.com/tokio-rs/tokio/issues/695

Using a new-type avoids forgetting to use the right API and forces opt-in to the ability of mocking time.

Reply:
hmm, that makes sense. I think it would be worth explaining that motivation in the docs, then.